src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -975,6 +975,7 @@ private ApiResponseSet alertToSet(Alert alert) {
 		map.put("id", String.valueOf(alert.getAlertId()));
 		map.put("pluginId", String.valueOf(alert.getPluginId()));
 		map.put("alert", alert.getAlert());
```

Comment:
This should also use getName() (to not use the deprecated method) and it might worth adding a comment like the one in the Alert.toPluginXML(String). I'd also add a XXX task tag to be more visible, for future removal.